### ISSUE TITLE
Debug Spew in test_autograd.py

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#8577]
See recent CI, which spews the following:

23:49:53 test_anomaly_detect_nan (__main__.TestAutograd) ... No forward pass information available.
23:49:53 Enable detect anomaly during forward pass for more informations.
23:49:53 No forward pass information available.
23:49:53 Enable detect anomaly during forward pass for more informations.
23:49:53 test_autograd.py:2328: DeprecationWarning: Please use assertRaisesRegex instead.
23:49:53   with self.assertRaisesRegexp(RuntimeError, "Function 'MyFuncBackward' returned nan values in its 0th output."):
23:49:53 ok

**Comment by goldsborough** [comment]
Should be a simple codemod I would think?

**Comment by SsnL** [comment]
This is weird because we have this line
https://github.com/pytorch/pytorch/blob/master/test/common.py#L484
Which py version is this?

**Comment by mruberry** [comment]
You can see this is the 3.6 and 3.5 builds. The 2.7.9 and 2.7 builds have a lesser version of the same issue:

05:06:01 test_anomaly_detect_nan (__main__.TestAutograd) ... No forward pass information available.
05:06:01 Enable detect anomaly during forward pass for more informations.
05:06:01 No forward pass information available.
05:06:01 Enable detect anomaly during forward pass for more informations.
05:06:01 ok

**Comment by SsnL** [comment]
Can you give us CI links for the two errors?

**Comment by mruberry** [comment]
Here's a link to the console log from a recent submission that passed all the tests:

https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-xenial-cuda9-cudnn7-py3-test2/1279/consoleFull

**Comment by mruberry** [comment]
As an aside from the immediate issue, may I suggest that the CI also fail if there's unexpected spew like this? I recently fixed another one of these spews with #8531.

**Comment by ezyang** [comment]
Yes, I think that should be doable. We can add logic to `setUp` and `tearDown` to check if there are any warnings and fail the test if so. Then, tests explicitly swallow warnings when relevant.

**Comment by ezyang** [comment]
I still see a lot of warning spew in test suites, so guessing we still haven't fixed this.
